src/Storage/Storage/DatalakeGen2/Cmdlet/NewAzDataLakeGen2Item.cs

```diff
@@ -0,0 +1,332 @@
+                SetBlobMetaData((CloudBlockBlob)destBlob, this.BlobMetadata, false);
+            };
+
+            await DataMovementTransferHelper.DoTransfer(() =>
+            {
+                return this.TransferManager.UploadAsync(filePath,
+                    blob,
+                    null,
+                    transferContext,
+                    this.CmdletCancellationToken);
+            },
+                data.Record,
+                this.OutputStream).ConfigureAwait(false);
+
+            // Set blob permission with umask, since create Blob API still not support them
+            SetBlobPermissionWithUMask((CloudBlockBlob)blob, this.Permission, this.Umask);
+
+            blob.FetchAttributes();
+            WriteDataLakeGen2Item(localChannel, (CloudBlockBlob)blob, taskId: data.TaskId);
+        }
+
+        /// <summary>
+        /// Set block blob permission with Umask after blob is created
+        /// Add this since XSCL don't have interface to set permission with umask in create blob
+        /// Won't set when both permission and umask are null/empty, and use server default behavior.
+        /// </summary>
+        /// <param name="blob">the blob object to set permission with umask</param>
+        /// <param name="permission">permission string to set, in format "rwxrwxrwx", default value is "rwxrwxrwx"</param>
+        /// <param name="umask">umask string to set, in format "rwxrwxrwx", default value is "----w-rwx"</param>
+        protected static void SetBlobPermissionWithUMask(CloudBlockBlob blob, string permission, String umask)
+        {
+            // Don't need set permission when both input permission and umask are null
+            if (string.IsNullOrEmpty(permission) && string.IsNullOrEmpty(umask))
+            {
+                return;
+            }
+
+            //Set the default value if one of permission or Umask is null
+            // Confirmed with feature team:
+            //  Default permission for files is 666 (default permission for directory is 777, this function is for file so use 666)
+            //  default umask is 027
+            if (string.IsNullOrEmpty(permission))
+            {
+                permission = "rw-rw-rw-";
+            }
+            if (string.IsNullOrEmpty(umask))
+            {
+                umask = "----w-rwx";
+            }
+
+            // Get the permission value to set, from input Permission and Umask
+            // The permission and umask string format is already checked with parameter ValidatePattern
+            string blobPermission = string.Empty;
+            for (int i = 0; i < permission.Length; i++)
+            {
+                if (umask[i] != '-')
+                {
+                    blobPermission += '-';
+                }
+                else
+                {
+                    blobPermission += permission[i];
+                }
+            }
```

Comment:
I prefer to refact this into a function.

Reply:
This only use in one place, and it might be removed late if XSCL support unmask in create file later, so I don’t prefer to make it in another function.